pkg/cloud/services/network/vpc.go

```diff
@@ -202,6 +202,12 @@ func (s *Service) createVPC() (*infrav1.VPCSpec, error) {
 func (s *Service) deleteVPC() error {
 	vpc := s.scope.VPC()
 
+	// Skip deletion if VPC is deleted already or not mentioned in .spec.vpc.id.
```

Comment:
When VPC is deleted, this field is not being set to nil.

Reply:
Done